### PR TITLE
fix: add missing stoplight dependencies

### LINKED ISSUE #733: Undeclared dependencies error under Yarn PnP
#### Describe the bug
[Yarn PnP](https://yarnpkg.com/features/pnp) needs all dependencies to be declared. This causes an error at runtime when the parser tries to access the following dependencies that are not declared in the [manifest](https://github.com/asyncapi/parser-js/blob/next-major/package.json#L44).

Example:
> Error: @asyncapi/parser tried to access @stoplight/types, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

The workaround is to add the undeclared dependencies to the user package's `.yarnrc.yml` file like so:

```yaml
packageExtensions:
  "@asyncapi/parser@*":
    dependencies:
      "@stoplight/types": "^13.9.1"
      "@stoplight/json": "^3.20.2"
      "@stoplight/spectral-ref-resolver": "^1.0.3"
      "@stoplight/json-ref-readers": "^1.2.2"
```

The solution is to add these as dependencies or peerDependencies (optional or otherwise).

#### How to Reproduce
Install and run the parser under [Yarn PnP](https://yarnpkg.com/features/pnp) package manager

#### Full errors with call stack:

<details>

<summary>@stoplight/types</summary>

```
Error: @asyncapi/parser tried to access @stoplight/types, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @stoplight/types
Required by: @asyncapi/parser@npm:2.0.0-next-major.15 (via /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/)

Require stack:
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/AsyncAPIInputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/InputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/AbstractGenerator.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/generate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/program.js
    at Function.require$$0.Module._resolveFilename (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22559:13)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22413:42)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/index.js:18:17)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Object.require$$0.Module._extensions..js (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22603:33)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22443:14)
```

Offending line seems to be this one: https://github.com/asyncapi/parser-js/blob/next-major/src/index.ts#L1
</details>

<details>
<summary>@stoplight/json</summary>

```
Error: @asyncapi/parser tried to access @stoplight/json, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @stoplight/json
Required by: @asyncapi/parser@npm:2.0.0-next-major.15 (via /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/utils/)

Require stack:
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/utils/mergeTraits.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/utils/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/v2/functions/channelParameters.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/v2/ruleset.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/v2/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/spectral.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/validate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parse.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parser.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/AsyncAPIInputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/InputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/AbstractGenerator.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/generate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/program.js
    at Function.require$$0.Module._resolveFilename (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22570:13)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22424:42)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/ruleset/utils/mergeTraits.js:4:16)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Object.require$$0.Module._extensions..js (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22614:33)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22454:14)
```

Offending line seems to be this one: https://github.com/asyncapi/parser-js/blob/next-major/src/ruleset/utils/mergeTraits.ts#L1
</details>

<details>
<summary>@stoplight/spectral-ref-resolver</summary>

```
Error: @asyncapi/parser tried to access @stoplight/spectral-ref-resolver, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @stoplight/spectral-ref-resolver
Required by: @asyncapi/parser@npm:2.0.0-next-major.15 (via /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/)

Require stack:
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/resolver.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/spectral.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/validate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parse.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parser.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/AsyncAPIInputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/InputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/AbstractGenerator.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/generate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/program.js
    at Function.require$$0.Module._resolveFilename (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22572:13)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22426:42)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/resolver.js:13:33)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Object.require$$0.Module._extensions..js (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22616:33)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22456:14)
```

Offending line seems to be this one: https://github.com/asyncapi/parser-js/blob/next-major/src/resolver.ts#L1
</details>

<details>
<summary>@stoplight/json-ref-readers</summary>

```
Error: @asyncapi/parser tried to access @stoplight/json-ref-readers, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.

Required package: @stoplight/json-ref-readers
Required by: @asyncapi/parser@npm:2.0.0-next-major.15 (via /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/)

Require stack:
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/resolver.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/spectral.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/validate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parse.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/parser.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/AsyncAPIInputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/InputProcessor.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/processors/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/AbstractGenerator.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/generators/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-modelina-npm-2.0.0-next.4-52848d6dc3-f26a0e0414.zip/node_modules/@asyncapi/modelina/lib/cjs/index.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/generate.js
- /Users/mg/WebstormProjects/mbh/asyncapi-generator/build/src/program.js
    at Function.require$$0.Module._resolveFilename (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22574:13)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22428:42)
    at Module.require (node:internal/modules/cjs/loader:1028:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.yarn/cache/@asyncapi-parser-npm-2.0.0-next-major.15-54e6ea0b35-2502272baa.zip/node_modules/@asyncapi/parser/cjs/resolver.js:15:28)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Object.require$$0.Module._extensions..js (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22618:33)
    at Module.load (node:internal/modules/cjs/loader:1004:32)
    at Function.require$$0.Module._load (/Users/mg/WebstormProjects/mbh/asyncapi-generator/.pnp.cjs:22458:14)

```

Offending line seems to be this one: https://github.com/asyncapi/parser-js/blob/next-major/src/resolver.ts#L2
</details>

### PR DESCRIPTION
**Pull request by woylie**
**Description**

This adds four stoplight dependencies that were missing, which caused Yarn 2/3 in pnp mode to return an error. I hope this is all that had to be done and doesn't cause any issues.

**Related issue(s)**
Resolves #733

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1760,9 +1832,9 @@
       }
     },
     "node_modules/@stoplight/types": {
-      "version": "13.12.0",
-      "resolved": "https://registry.npmjs.org/@stoplight/types/-/types-13.12.0.tgz",
-      "integrity": "sha512-lpyB8b5cNcg/CyQ9W7Q2BxPkHhBA8+9byviGT6uL2Ok/5iao0poCnFUH++Cd7VN1/RaXrv0LT9fyOg4fJIl0iA==",
+      "version": "14.0.0",
```
Comment (smoya) [inline package-lock.json:1835]: This increases the major version. We need to either ensure nothing affects us from the breaking change in their changelog or rather downgrade to the previous major.

Cc @magicmatatjahu so you are aware

Comment (woylie) [reply package-lock.json:1835]: My bad, let me downgrade.

The breaking changelog entry is: `http-spec: IHttpService['security'] is a 2D array now`. Not sure whether that's relevant here, `IHttpService` doesn't appear in the code.

**Comment by smoya** [comment]
@woylie `npm run install` command is failing with: 
```
npm verb stack @stoplight/json-ref-readers: No matching version found for @stoplight/json-ref-readers@3.1.5.
npm verb stack     at module.exports (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/npm-pick-manifest/lib/index.js:209:23)
npm verb stack     at RegistryFetcher.manifest (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/pacote/lib/registry.js:125:22)
npm verb stack     at async Arborist.[nodeFromEdge] (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1061:19)
npm verb stack     at async Arborist.[buildDepStep] (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:930:11)
npm verb stack     at async Arborist.buildIdealTree (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:211:7)
npm verb stack     at async Promise.all (index 1)
npm verb stack     at async Arborist.reify (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:1[54](https://github.com/asyncapi/parser-js/actions/runs/6535668041/job/17758980054?pr=882#step:8:55):5)
npm verb stack     at async Install.exec (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/lib/commands/install.js:145:5)
npm verb stack     at async module.exports (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/lib/cli.js:78:5)
npm verb cwd /home/runner/work/parser-js/parser-js
npm verb Linux 6.2.0-1012-azure
npm verb node v16.20.2
npm verb npm  v8.19.4
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @stoplight/json-ref-readers@3.1.5.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
```

See https://github.com/asyncapi/parser-js/actions/runs/6535668041/job/17758980054?pr=882#step:8:49

**Comment by woylie** [comment]
> @woylie `npm run install` command is failing with:
> 
> ```
> npm verb stack @stoplight/json-ref-readers: No matching version found for @stoplight/json-ref-readers@3.1.5.
> npm verb stack     at module.exports (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/npm-pick-manifest/lib/index.js:209:23)
> npm verb stack     at RegistryFetcher.manifest (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/pacote/lib/registry.js:125:22)
> npm verb stack     at async Arborist.[nodeFromEdge] (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1061:19)
> npm verb stack     at async Arborist.[buildDepStep] (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:930:11)
> npm verb stack     at async Arborist.buildIdealTree (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:211:7)
> npm verb stack     at async Promise.all (index 1)
> npm verb stack     at async Arborist.reify (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/node_modules/@npmcli/arborist/lib/arborist/reify.js:1[54](https://github.com/asyncapi/parser-js/actions/runs/6535668041/job/17758980054?pr=882#step:8:55):5)
> npm verb stack     at async Install.exec (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/lib/commands/install.js:145:5)
> npm verb stack     at async module.exports (/opt/hostedtoolcache/node/16.20.2/x64/lib/node_modules/npm/lib/cli.js:78:5)
> npm verb cwd /home/runner/work/parser-js/parser-js
> npm verb Linux 6.2.0-1012-azure
> npm verb node v16.20.2
> npm verb npm  v8.19.4
> npm ERR! code ETARGET
> npm ERR! notarget No matching version found for @stoplight/json-ref-readers@3.1.5.
> npm ERR! notarget In most cases you or one of your dependencies are requesting
> npm ERR! notarget a package version that doesn't exist.
> ```
> 
> See https://github.com/asyncapi/parser-js/actions/runs/6535668041/job/17758980054?pr=882#step:8:49

I shouldn't stay up so late.

**Review by smoya — APPROVED** [review]
LGTM! 🚀🌔

**Comment by smoya** [comment]
@magicmatatjahu would you like to take a look just in case? I gave my +1

**Comment by jonaslagoni** [comment]
Dont see a reason to wait, its not like it will break anything 🙂

/rtm

**Pull request merged**